packages/SwingSet/docs/dynamic-vats.md

```diff
@@ -160,10 +160,11 @@ When the vat halted due to a metering fault, `error` will be a `RangeError` with
 
 ## Upgrade
 
-Dynamic vats can be upgraded to use a new source code bundle. Most vat state is discarded, however "durable" collections are retained for use by the replacement version. For full details, see docs/vat-upgrade.md .
+Dynamic vats can be upgraded to use a new source code bundle. Most vat state is discarded, however "durable" collections are retained for use by the replacement version. For full details, see [vat-upgrade.md](./vat-upgrade.md).
 
-The upgrade process is triggered through the admin node, and requires a bundlecap for the new source. (Note that "upgrading" to the same source bundle as the original vat is valid, and can serve as a sort of "spring cleaning").
+The upgrade process is triggered through the vat's "adminNode" control facet, and requires specifying the new source bundle. (Note that a "null upgrade" that re-uses the original bundle is valid, and a legitimate approach to deleting accumulated state).
```

Comment:
The API call specifically requires a bundlecap, not a bundlel there is no way to upgrade a vat to a naked bundle. Maybe change this to "requires specifying the new source code (as a BundleCap)" ?